packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts

```diff
@@ -701,9 +708,15 @@ export class CurrentActivityPrinter extends ActivityPrinterBase {
     return '[' + color(fullChars + partialChar) + filler + `] (${this.resourcesDone}/${this.resourcesTotal})`;
   }
 
+  private failureReason(activity: StackActivity) {
+    const resourceStatusReason = activity.event.ResourceStatusReason ?? '';
+    const logicalResourceId = activity.event.LogicalResourceId ?? '';
+    return resourceStatusReason.includes('The following hook(s) failed')? resourceStatusReason + ' : ' + this.hookFailureMap.get(logicalResourceId) ?? '' : resourceStatusReason;
```

Comment:
Is it guaranteed that this text won't change at some point? Is there an error code or something more stable we can look at?

Reply:
Hey @ericzbeard , there is no error code or other property in event that can determine if a resource failed because of hook or not.  One another proposal I can have here is to store hookFailureMap as 
{
  “resource1": {
    “hookname1": "reason1",
    “hookname2": "reason2"
  },
  “resource2": {
    "hookname3”: "reason3",
    “hookname4: "reason4"
  }
}

Instead of using "The following hook(s) failed" to determine if a resource failed because of hook or not. We can use "^[a-zA-Z0-9]{2,64}::[a-zA-Z0-9]{2,64}::[a-zA-Z0-9]{2,64}$" which is regex of hookName to see if resource failed because of specific hook. (reference: https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/hooks-structure.html)